src/main/java/io/jenkins/plugins/appcenter/task/internal/UploadAppToResourceTask.java

```diff
@@ -60,7 +61,7 @@
 
         final CompletableFuture<UploadRequest> future = new CompletableFuture<>();
 
-        final File file = new File(filePath.child(pathToApp).getRemote());
+        final File file = remoteFileUtils.getRemoteFile(pathToApp);
```

Comment:
I've extracted this into a separate class because then we can inject a mocked version for testing and more easily fake the characteristics of the file. For example we can fake the `length` property to trigger the chunked upload flow without needing to commit a 256MB file to the repo.